### ISSUE TITLE
Syntax error on `# type: Any` after a module import

### ISSUE THREAD

**Issue by Pierre-Sassoulas** [pylint-dev/pylint#8655]
Pylint also doesn't like `# type: Any` comments after a module import:

```python
from collections import abc  # type: Any
```

```
E0001: Parsing failed: 'invalid syntax (<unknown>, line 1)' (syntax-error)
```

```
$ pylint --version
pylint 2.16.2
astroid 2.14.2
Python 3.11.0 (main, Feb 21 2023, 11:59:45) [GCC 12.2.0]
```

_Originally posted by @yilei in https://github.com/pylint-dev/pylint/issues/3556#issuecomment-1449030317_

**Comment by mbyrnepr2** [comment]
I think this is the same root-cause as #3556 (`python -m ast example.py` on that example is also a SyntaxError).

Generally there is a SyntaxError with code corresponding to the [ast module](https://docs.python.org/3/library/ast.html) class with no `type_comment` parameter.
For example, we can reproduce the issue with:

`cat example.py:`
```python
if True: # type: Any
    print("True")
```

`pylint example.py`
py:1:18: E0001: Parsing failed: 'invalid syntax (<unknown>, line 1)' (syntax-error)

`python -m ast example.py`
```python
  File "example.py", line 1
    if True: # type: Any
                     ^^^
SyntaxError: invalid syntax
```
